modules/src/mock_client/state.rs

```diff
@@ -61,16 +67,24 @@ impl From<MockClientState> for AnyClientState {
     }
 }
 
-impl TryFromRaw for MockClientState {
-    type RawType = ibc_proto::ibc::mock::ClientState;
+impl TryFrom<RawMockClientState> for MockClientState {
     type Error = Error;
 
-    fn try_from(raw: Self::RawType) -> Result<Self, Self::Error> {
-        let raw_header = raw
-            .header
-            .ok_or_else(|| Kind::InvalidRawClientState.context("missing header"))?;
+    fn try_from(raw: RawMockClientState) -> Result<Self, Self::Error> {
+        Ok(MockClientState(raw.header.unwrap().try_into()?))
+    }
+}
 
-        Ok(Self(MockHeader::try_from(raw_header)?))
+impl From<MockClientState> for RawMockClientState {
+    fn from(value: MockClientState) -> Self {
+        RawMockClientState {
+            header: Some(ibc_proto::ibc::mock::Header {
+                height: Some(ibc_proto::ibc::client::Height {
+                    epoch_number: 0,
```

Comment:
There's quite a few places where we decode into Some(Height...). I guess it will make sense to make Height a domain type as well & reduce duplicate code.

Reply:
Yes! The purpose here was to emerge all the current implementation state. Optimization is on the table in a followup PR!